src/sentry/migrations/0335_cleanup_old_savedsearches.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 2.2.28 on 2022-10-29 07:47
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def cleanup_savedsearch(apps, schema_editor):
+    SavedSearch = apps.get_model("sentry", "SavedSearch")
+
+    # This removes all old said searches that have associated project_id.
+    #
+    # We no longer provide an API to query searches with a project_id and
+    # instead provide organization level saved searches. There are MANY old
+    # saved searches since
+    #
+    # For more contextual history of this change see [0] (you must work at
+    # Sentry, sorry)
+    #
+    # [0]: https://www.notion.so/sentry/Saved-Search-cleanup-d2c2ae68b2174453a63ec19003ae17ae
+    for search in RangeQuerySetWrapperWithProgressBar(SavedSearch.objects.all()):
+
+        # There shouldn't be any searches with a project ID AND a
+        # organization_id, but just to be safe, let's include the check here
+        if search.project_id is not None and search.organization_id is None:
```

Comment:
Couldn't these be conditions on the original query? Both columns are in indexes so they should be efficient to use in where conditions.

Reply:
The use of `RangeQuerySetWrapperWithProgressBar` tends to break index usage, since it orders/filters by the `id` column.

That said, the advice to always use `.all` came from Matt and I'm not sure if it's totally necessary in most cases. The reasoning there was that using `.all()` results in more low cost queries that run over a longer period of time, and so reduces the load on Postgres, vs doing potentially more expensive sequential scans over a shorter period of time.